jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1023,26 +1024,22 @@ typedef double ecma_number_t;
  */
 typedef struct
 {
-  /** Number of elements in the collection */
-  ecma_length_t unit_number;
-
-  /** Compressed pointer to first chunk with collection's data */
-  jmem_cpointer_t first_chunk_cp;
-
-  /** Compressed pointer to last chunk with collection's data */
-  jmem_cpointer_t last_chunk_cp;
+  jmem_cpointer_t first_chunk_cp; /**< compressed pointer to first chunk with collection's data */
+  jmem_cpointer_t last_chunk_cp; /** Compressed pointer to last chunk with collection's data */
+  ecma_length_t item_count; /**< number of items in the collection */
 } ecma_collection_header_t;
 
+/**
+ * String is not a valid array index.
+ */
+#define ECMA_COLLECTION_CHUNK_ITEMS 5
+
 /**
  * Description of non-first chunk in a collection's chain of chunks
  */
 typedef struct
 {
-  /** Characters */
-  lit_utf8_byte_t data[ sizeof (uint64_t) - sizeof (jmem_cpointer_t) ];
-
-  /** Compressed pointer to next chunk */
-  jmem_cpointer_t next_chunk_cp;
+  ecma_value_t items[ECMA_COLLECTION_CHUNK_ITEMS + 1];
```

Comment:
Could you please add a comment here that clarifies why the +1 value is added and what purpose it serves.